### PR TITLE
Count CPU flash attention FLOPs

### LINKED ISSUE #195460: FlopCounterMode counts 0 FLOPs for CPU scaled_dot_product_attention (flop_registry lacks aten::_scaled_dot_product_flash_attention_for_cpu)
### 🐛 Describe the bug

`torch.utils.flop_counter.FlopCounterMode` reports **0 FLOPs** for `scaled_dot_product_attention` on CPU, because the CPU default path dispatches to `aten::_scaled_dot_product_flash_attention_for_cpu`, which has no entry in `flop_registry`. The CUDA counterpart (`aten::_scaled_dot_product_flash_attention`) is registered, so the same model reports correct attention FLOPs on GPU and silently drops them on CPU. Note this affects eval AND train mode: with dropout 0.0 the CPU dispatcher still selects the fused `_for_cpu` kernel.

Repro (torch 2.13.0):

```python
import torch
from torch.utils.flop_counter import FlopCounterMode, flop_registry

q = torch.randn(1, 12, 64, 64)  # (B, H, S, D)
with FlopCounterMode(display=False) as m:
    torch.nn.functional.scaled_dot_product_attention(q, q, q)
print(m.get_total_flops())  # 0  (expected: 12_582_912)

print(torch.ops.aten._scaled_dot_product_flash_attention_for_cpu in flop_registry)  # False
print(torch.ops.aten._scaled_dot_product_flash_attention in flop_registry)          # True
```

Dispatch confirmation: a `TorchDispatchMode` spy on the call above sees `aten._scaled_dot_product_flash_attention_for_cpu.default`.

Expected count for the shape above: QK^T + AV = 2 x (2 x B x H x S^2 x D) = 12,582,912 FLOPs per call (150,994,944 across a 12-layer stack), matching what `sdpa_flop` computes for the registered CUDA overloads.

Suggested fix: register the `_for_cpu` overload (and its backward, `aten::_scaled_dot_product_flash_attention_for_cpu_backward`, alongside the existing `sdpa_backward` registrations) with the existing `sdpa_flop` / `sdpa_backward_flop` formulas in `torch/utils/flop_counter.py`.

Found while cross-checking FLOP counts against dispatcher-level op logs in [torchlens](https://github.com/johnmarktaylor91/torchlens); happy to send a PR if useful.

### Versions

Reproduced on torch 2.13.0 (cu130 build, CPU execution). The `_for_cpu` overload is absent from `flop_registry` on every version we checked back to 2.1.

### PR DESCRIPTION
**Pull request by wei-core**
Human commentary from the contributor:

CPU SDPA performs real attention computation, so reporting zero FLOPs makes `FlopCounterMode` inaccurate for CPU attention workloads. I reviewed and understand the two-file change: it registers the CPU flash-attention forward and backward operators with the existing SDPA FLOP formulas and adds regression coverage for the expected forward and backward counts. I also reviewed the reported test results and approve this change.

> AI-assisted implementation summary, reviewed line by line by the contributor:
>
> `FlopCounterMode` has formulas for fused scaled-dot-product attention, but
> its registry omits the CPU-specific flash-attention forward and backward
> operators. Consequently, CPU `scaled_dot_product_attention` calls are
> silently reported as zero FLOPs.
>
> Register `_scaled_dot_product_flash_attention_for_cpu` and
> `_scaled_dot_product_flash_attention_for_cpu_backward` with the existing
> SDPA formulas. Their leading tensor arguments match the formulas already
> used for the other fused SDPA backends.
>
> Add a CPU regression that checks the forward count and then verifies that
> backward adds the expected backward count.
>
> Fixes #195460.
>
> Test plan:
>
> - Loaded the modified local `torch.utils.flop_counter` into the existing
>   PyTorch test runner and ran
>   `TestFlopCounter.test_sdpa_cpu` and
>   `TestFlopCounter.test_sdpa_flop_count_gqa`: 2 tests passed.
> - `git diff --check`: passed.
> - `uvx --python 3.10 lintrunner@0.12.7 -a`: attempted, but the linter
>   bootstrap reports `Windows/Windows-AMD64` as unsupported and cannot
>   initialize in this Windows sparse checkout. It produced no file-specific
>   findings for this change.
>
> AI assistance disclosure: the implementation and quoted PR summary were
> prepared with OpenAI Codex. Codex ran the tests listed above; the contributor
> reviewed the code line by line and is responsible for the submitted change.

### CODE REVIEW

**Comment by wei-core** [comment]
@drisspg Thanks for triaging this. I rebased the PR onto current `main` and resolved the conflict caused by the recent `test_flop_counter.py` CPU/CUDA test-class split. The PR is mergeable again, and the functional diff is unchanged (4 registration lines plus the 18-line CPU forward/backward regression).

Revalidated the rebased head (`b934c5d`):

- `TestFlopCounter.test_sdpa_cpu`
- `TestFlopCounter.test_sdpa_flop_count_gqa`
- Result: 2 tests passed
- `git diff --check`: passed

The new Lint, BC Lint, pull, and docs-build workflow runs are currently marked `action_required` pending approval for this fork PR. Could you please approve the workflows and take another look when convenient?

**Review by drisspg — APPROVED** [review]
